solidity/contracts/deposit/DepositRedemption.sol

```diff
@@ -31,13 +31,10 @@ library DepositRedemption {
     /// @notice     Pushes signer fee to the Keep group by transferring it to the Keep address.
     /// @dev        Approves the keep contract, then expects it to call transferFrom.
     function distributeSignerFee(DepositUtils.Deposit storage _d) internal {
-        address _tbtcTokenAddress = _d.TBTCToken;
-        TBTCToken _tbtcToken = TBTCToken(_tbtcTokenAddress);
-
         IBondedECDSAKeep _keep = IBondedECDSAKeep(_d.keepAddress);
 
-        _tbtcToken.approve(_d.keepAddress, _d.signerFee());
-        _keep.distributeERC20Reward(_tbtcTokenAddress, _d.signerFee());
+        _d.tbtcToken.approve(_d.keepAddress, _d.signerFee());
+        _keep.distributeERC20Reward(address(_d.tbtcToken), _d.signerFee());
```

Comment:
This feels like it should propagate to keep, insofar as `distributeERC20Reward` should take an `IERC20`?

Reply:
Don't think I follow :s

Reply:
Sorry, I mean `distributeERC20Reward` should declare its first parameter as an `IERC20` rather than an `address`, and then we wouldn't need this cast, right? Obviously we'd have to make that change in keep-ecdsa as well.

Reply:
I'm not sure about this. At this point we're losing any compiler advantages anyway, no?

Reply:
Hmmm… How so? The interface would require `IERC20`, and the Keep side also implements that interface. Even though there isn't a build of the whole system where the compiler will catch stuff, the interfaces allow us to catch those same issues at compile time.

Reply:
Alright, let's change it. Needs `keep-ecdsa` PR.

Reply:
moving to separate PR